helpers/libraries/serveHelpers.js

```diff
@@ -29,39 +98,157 @@ module.exports = {
     // send the file
     res.status(200).sendFile(filePath, options);
   },
-  validateClaimId (name, claimId) {
+  getClaimIdByShortUrl (name, shortUrl) {
+    const deferred = new Promise((resolve, reject) => {
+      lbryApi.getClaimsList(name)
+      .then(({ claims }) => {
+        const regex = new RegExp(`^${shortUrl}`);
+        logger.debug('regex:', regex);
+        const filteredClaimsList = claims.filter(claim => {
+          return regex.test(claim.claim_id);
+        });
+        logger.debug('filtered claims list', filteredClaimsList);
+        switch (filteredClaimsList.length) {
+          case 0:
+            reject(new Error('That is an invalid short url'));
+            break;
+          case 1:
+            resolve(filteredClaimsList[0].claim_id);
+            break;
+          default:
+            const sortedClaimsList = filteredClaimsList.sort((a, b) => {
+              return a.height > b.height;
```

Comment:
Technically two claims could have the same height, though this is rather unlikely.